src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+        # Then get the circumradius of the region from that point and the farthest point on
+        # the object from this point. If the maximum distance is greater than the circumradius,
+        # return False.
+
+        # Get a candidate point from the rgion mesh. If the center of mass of the region is in the mesh use that.
+        # Otherwise try to sample a point as a candidate, skipping this pass if the sample fails.
+        if self.containsPoint(Vector(*self.mesh.bounding_box.center_mass)):
+            reg_candidate_point = Vector(*self.mesh.bounding_box.center_mass)
+        elif len(samples:=trimesh.sample.volume_mesh(self.mesh, self.num_samples)) > 0:
+            reg_candidate_point = Vector(*samples[0])
+        else:
+            reg_candidate_point = None
+
+        if reg_candidate_point is not None:
+            # Calculate circumradius of the region from the candidate_point
+            reg_circumradius = numpy.max(numpy.linalg.norm(self.mesh.vertices - reg_candidate_point, axis=1))
+
+            # Calculate maximum distance to the object.
+            obj_max_distance = numpy.max(numpy.linalg.norm(obj.occupiedSpace.mesh.vertices - reg_candidate_point, axis=1))
+
+            if obj_max_distance > reg_circumradius:
+                return False
+
+        # PASS 5
+        # If the difference between the object's region and this region is empty,
+        # i.e. obj_region - self_region = EmptyRegion, that means the object is
+        # entirely contained in this region.
+        diff_region = obj.occupiedSpace.difference(self)
+
+        return isinstance(diff_region, EmptyRegion)
+
+    def containsRegionInner(self, reg, tolerance):
+        if tolerance != 0:
+            warnings.warn("Nonzero tolerances are ignored for containsRegionInner on MeshVolumeRegion")
+
+        if isinstance(reg, MeshVolumeRegion):
+            diff_region = reg.difference(self)
+
+            return isinstance(diff_region, EmptyRegion)
+
+        raise NotImplementedError
+
+    # Composition methods #
+    @lru_cache(maxsize=None)
+    def intersect(self, other, triedReversed=False):
+        """ Get a `Region` representing the intersection of this region with another.
+
+        This function handles intersection computation for `MeshVolumeRegion` with:
+        * `MeshVolumeRegion`
+        * `PolygonalFootprintRegion`
+        * `PolygonalRegion`
+        * `PathRegion`
+        * `PolylineRegion`
+        """
+        # If one of the regions isn't fixed fall back on default behavior
+        if needsSampling(self) or needsSampling(other):
+            return super().intersect(other, triedReversed)
+
+        if isinstance(other, MeshVolumeRegion):
+            # Other region is a mesh volume. We can extract the mesh to perform boolean operations on it
+            other_mesh = other.mesh
+
+            # Compute intersection using Trimesh (CalledProcessError usually means empty intersection for OpenSCAD)
```

Comment:
This seems fragile. Is there any way to tell that this wasn't the user hitting Ctrl-C or SCAD aborting in some other way?

Reply:
It is, but from what I can see this is fundamental to the way Trimesh calls OpenSCAD.

Reply:
Should be addressed by this PR: https://github.com/mikedh/trimesh/pull/1940

Once it's merged in we can remove the catches from Scenic and let CalledProcessErrors propagate upwards.